src/main/java/jp/co/soramitsu/sora/crypto/algorithms/SignatureSuiteRegistry.java

```diff
@@ -0,0 +1,45 @@
+package jp.co.soramitsu.sora.crypto.algorithms;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class SignatureSuiteRegistry {
+
+  private static Map<String, RawSignatureStrategy> algorithms = new HashMap<>();
+
+  private SignatureSuiteRegistry() {
+  }
+
+  public static void register(String key, RawSignatureStrategy val) {
+    algorithms.put(key, val);
+  }
+
+  public static void deregister(String key) {
+    algorithms.remove(key);
+  }
+
+  public static RawSignatureStrategy get(String key) throws InvalidAlgorithmException {
+    // find appropriate digital signature algorithm
+    if (!SignatureSuiteRegistry.has(key)) {
+      throw new InvalidAlgorithmException(key + " signature suite is not implemented");
+    }
+
+    return algorithms.get(key);
+  }
+
+  public static boolean has(String key) {
+    return algorithms.containsKey(key);
+  }
+
+  public static void clear() {
+    algorithms.clear();
+  }
+
+  public static class InvalidAlgorithmException extends Exception {
```

Comment:
Semantics mismatch, I suggest using name like "RawStrategyNotFound" or "NoSuchStrategy"
Also, consider having key used to retrieve `RawSignatureStrategy` as an argument of constructor of an exception, which will form an exception message itself, using it, like:

```java
public static class NoSuchStrategy extends Exception {
  public NoSuchStrategy(String key) {
    super(key + " signature suite is not implemented");
  }
}
```

While this is ok, do we actually need to throw an exception, or return an `Optional` instead and let the method caller decide what to do, if the strategy wasn't found, latter is usually more convenient